compaction/compaction.cc

```diff
@@ -1242,15 +1251,29 @@ class scrub_compaction final : public regular_compaction {
         flat_mutation_reader_v2 _reader;
         mutation_fragment_stream_validator _validator;
         bool _skip_to_next_partition = false;
+        std::optional<compaction_task_stats_ptr> _cts;
 
     private:
+        void register_validation_error() {
+            if (_scrub_mode == compaction_type_options::scrub::mode::segregate && _cts) {
```

Comment:
Can _cts be disengaged in `scrub::mode::segregate`?
If not, there's no point in testing that.

Reply:
Alternatively, you can just test `if (_cts)` and count without having to rely on the scrub::mode.
It's better than sprinkling the scrub logic all over the place.

Reply:
... and I think you meant `scrub::mode::validate` here, no? :)